web/src/lib/stores/assets-store.svelte.ts

```diff
@@ -1335,12 +1446,111 @@ export class AssetStore {
     }
   }
 
-  async getNextAsset(asset: AssetResponseDto): Promise<AssetResponseDto | undefined> {
-    let bucket = await this.#getBucketInfoForAsset(asset);
+  async getClosestAssetToDate(date: DateTime) {
+    let bucket = this.#findBucketForDate(date);
     if (!bucket) {
       return;
     }
+    await this.loadBucket(bucket.bucketDate, { cancelable: false });
+    const asset = bucket.findClosest(date);
+    if (asset) {
+      return asset;
+    }
+
+    let bucketIndex = this.buckets.indexOf(bucket) + 1;
+    while (bucketIndex < this.buckets.length) {
+      bucket = this.buckets[bucketIndex];
+      await this.loadBucket(bucket.bucketDate);
+      const next = bucket.dateGroups[0]?.intersetingAssets[0]?.asset;
+      if (next) {
+        return next;
+      }
+      bucketIndex++;
+    }
+  }
+
+  async getNextAsset(
+    idable: { id: string },
```

Comment:
This name is still confusing me. I'd just `{ id }: { id: string }` if you don't want to consider it anything more than an id.